diff -ruN tar-1.26/gnu/argp-parse.c tar-1.26-modified/gnu/argp-parse.c
--- tar-1.26/gnu/argp-parse.c	2011-03-12 04:14:26.000000000 -0500
+++ tar-1.26-modified/gnu/argp-parse.c	2012-04-30 18:18:05.000000000 -0400
@@ -218,6 +218,8 @@
   void *hook;
 };
 
+extern error_t parse_opt (int key, char *arg, struct argp_state *state);
+
 /* Call GROUP's parser with KEY and ARG, swapping any group-specific info
    from STATE before calling, and back into state afterwards.  If GROUP has
    no parser, EBADKEY is returned.  */
@@ -231,7 +233,15 @@
       state->input = group->input;
       state->child_inputs = group->child_inputs;
       state->arg_num = group->args_processed;
-      err = (*group->parser)(key, arg, state);
+      //err = (*group->parser)(key, arg, state);
+      if (group->parser == argp_default_parser)
+        err = argp_default_parser(key, arg, state);
+      else if (group->parser == argp_version_parser)
+        err = argp_version_parser(key, arg, state);
+      else if (group->parser == parse_opt)
+        err = parse_opt(key, arg, state);
+      else
+        err = (*group->parser)(key, arg, state);
       group->hook = state->hook;
       return err;
     }
diff -ruN tar-1.26/src/tar.c tar-1.26-modified/src/tar.c
--- tar-1.26/src/tar.c	2010-10-24 14:07:55.000000000 -0400
+++ tar-1.26-modified/src/tar.c	2012-04-30 18:18:26.000000000 -0400
@@ -1366,7 +1366,7 @@
 
 #define TAR_SIZE_SUFFIXES "bBcGgkKMmPTtw"
 
-static error_t
+error_t
 parse_opt (int key, char *arg, struct argp_state *state)
 {
   struct tar_args *args = state->input;
